cmd/snap/cmd_known.go

```diff
@@ -110,9 +118,20 @@ func (x *cmdKnown) Execute(args []string) error {
 
 	var assertions []asserts.Assertion
 	var err error
-	if x.Remote {
+	switch {
+	case x.Remote && !x.Direct:
+		// --remote will query snapd
+		assertions, err = x.client.Known(string(x.KnownOptions.AssertTypeName), headers, &client.KnownOptions{Remote: true})
+		// if snapd is unavailable automatically fallback
+		var connErr client.ConnectionError
+		if xerrors.As(err, &connErr) {
+			assertions, err = downloadAssertion(string(x.KnownOptions.AssertTypeName), headers)
+		}
+	case x.Direct:
+		// --direct will always go direct, with or without --remote
 		assertions, err = downloadAssertion(string(x.KnownOptions.AssertTypeName), headers)
-	} else {
+	default:
+		// default is to look only local
 		assertions, err = x.client.Known(string(x.KnownOptions.AssertTypeName), headers, nil)
```

Comment:
The `--direct` case is covered by tests, but this line isn't hit by any test case.

Reply:
Nice catch, added now.